Rubberduck.Interaction/IMessageBox.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using Forms = System.Windows.Forms;
+
+namespace Rubberduck.Interaction
+{
+    public interface IMessageBox
+    {
+        void Notify(string text);
+        void NotifyError(string text, string caption);
+        void NotifyWarn(string text, string caption);
+        bool Question(string text, string caption);
+        bool Prompt(string text, string caption);
+        bool ConfirmYN(string text, string caption);
+        bool ConfirmYN(string text, string caption, bool suggestion);
+        bool? Confirm(string text, string caption, bool? suggestion);
+    }
+
+    public class MessageBox : IMessageBox
+    {
+        public void Notify(string text)
+        {
+            Forms.MessageBox.Show(text);
+        }
+
+        public void NotifyError(string text, string caption)
+        {
+            Forms.MessageBox.Show(text, caption, Forms.MessageBoxButtons.OK, Forms.MessageBoxIcon.Error);
+        }
+
+        public void NotifyWarn(string text, string caption)
+        {
+            Forms.MessageBox.Show(text, caption, Forms.MessageBoxButtons.OK, Forms.MessageBoxIcon.Exclamation);
+        }
+
+        public bool Question(string text, string caption)
+        {
+            return Forms.MessageBox.Show(text, caption, Forms.MessageBoxButtons.YesNo, Forms.MessageBoxIcon.Question) == Forms.DialogResult.Yes;
+        }
+
+        // FIXME check whether to merge with Question
+        public bool Prompt(string text, string caption)
+        {
+            return Forms.MessageBox.Show(text, caption, Forms.MessageBoxButtons.YesNo, Forms.MessageBoxIcon.Information) == Forms.DialogResult.Yes;
+        }
+
+        public bool ConfirmYN(string text, string caption)
+        {
+            return Forms.MessageBox.Show(text, caption, Forms.MessageBoxButtons.YesNo, Forms.MessageBoxIcon.Exclamation) == Forms.DialogResult.Yes;
+        }
+
+        public bool ConfirmYN(string text, string caption, bool suggestion)
+        {
+            return Forms.MessageBox.Show(text, caption, Forms.MessageBoxButtons.YesNo, Forms.MessageBoxIcon.Exclamation, suggestion ? Forms.MessageBoxDefaultButton.Button1 : Forms.MessageBoxDefaultButton.Button2) == Forms.DialogResult.Yes;
+        }
+
+        public bool? Confirm(string text, string caption, bool? suggestion)
```

Comment:
FWIW, instead of a `bool?` it seems better to wrap the `DialogResult` so that we can have more readable code like `var response = Confirm(...);` => `if(response = DialogResult.Cancel) ....`

Reply:
@bclothier meh. `if _msgbox.Confirm(...) {...}` is just as expressive IMO; would you know whether to check for `DialogResult.Yes` or `DialogResult.OK` if you weren't providing arguments telling you what the buttons are going to be? A Boolean makes it much clearer what the possible returned values are, provided that the caller is testing for the positive result, i.e. assuming that "not no" means "yes" would be a mistake... might be worth noting in the XML-doc summary 😉

Reply:
This works for the `bool` where there's only 2 results but with `bool?`, there's now 3 possible results. What does `null` means then? As it is, I'd have to look at the source code to see that `null` meant the user cancelled out and not some silly third button like `Abort`. In this case, I'd just have a custom `enum` allowing only `Yes`, `No`, `Cancel` -- no need to deal with `OK` or `Abort` or `null`.

Reply:
As mentioned in chat: this was just the quick-and-dirty fix to get the Forms API out of the IMessageBox interface. It's not really at the core of this PR. If you want a proper fix for it, I'd suggest opening a follow-up issue and assigning that to me.